Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryILLPreprocess.py

```diff
@@ -397,67 +346,110 @@ def _foregroundWidths(self):
+        theta = ws.spectrumInfo().twoTheta(mindex) / 2.
+        linePosition = 0.0
+        if not self.getProperty(Prop.INPUT_WS).isDefault:
+            inputWS = self.getProperty(Prop.INPUT_WS).value
+            if inputWS.run().hasProperty(common.SampleLogs.LINE_POSITION):
+                linePosition = inputWS.run().getLogData(common.SampleLogs.LINE_POSITION).value
+        if not self.getProperty(Prop.LINE_POSITION_INPUT).isDefault and linePosition == 0.0:
+            linePosition = self.getProperty(Prop.LINE_POSITION_INPUT).value
+        else:
+            # Fit peak position
+            peakWSName = self._names.withSuffix('peak')
+            args = {'InputWorkspace': ws, 'OutputWorkspace': peakWSName, 'EnableLogging': self._subalgLogging}
+            if not self.getProperty(Prop.XMIN).isDefault:
+                xmin = self.getProperty(Prop.XMIN).value
+                args['RangeLower'] = common.inTOF(xmin, l1, l2, theta)
+            if not self.getProperty(Prop.XMAX).isDefault:
+                xmax = self.getProperty(Prop.XMAX).value
+                args['RangeUpper'] = common.inTOF(xmax, l1, l2, theta)
+            if not self.getProperty(Prop.START_WS_INDEX).isDefault:
+                args['StartWorkspaceIndex'] = self.getProperty(Prop.START_WS_INDEX).value
+            if not self.getProperty(Prop.END_WS_INDEX).isDefault:
+                args['EndWorkspaceIndex'] = self.getPropertyValue(Prop.END_WS_INDEX).value
+            FindReflectometryLines(**args)
+            linePosition = mtd[peakWSName].readY(0)[0]
+            self._cleanup.cleanup(peakWSName)
+        return ws, linePosition
+
+    def _addSampleLogInfo(self, ws, linePosition):
+        """Add foreground indices and linePosition to the sample logs, names start with reduction."""
+        # Add the fractional workspace index of the beam position to the sample logs of ws.
+        run = ws.run()
+        run.addProperty(common.SampleLogs.LINE_POSITION, float(linePosition), True)
+        # Add foreground start and end workspace indices to the sample logs of ws.
+        hws = self._foregroundWidths()
+        beamPosIndex = int(numpy.rint(linePosition))
+        if beamPosIndex > 255:
+            beamPosIndex = 255
+            self.self.log().warning('Is it a monitor spectrum?')
+        sign = self._workspaceIndexDirection(ws)
+        startIndex = beamPosIndex - sign * hws[0]
+        endIndex = beamPosIndex + sign * hws[1]
+        if startIndex > endIndex:
+            endIndex, startIndex = startIndex, endIndex
+        run.addProperty(common.SampleLogs.FOREGROUND_START, int(startIndex), True)
+        run.addProperty(common.SampleLogs.FOREGROUND_CENTRE, int(beamPosIndex), True)
+        run.addProperty(common.SampleLogs.FOREGROUND_END, int(endIndex), True)
+        # Add two theta to the sample logs of ws.
+        if not self.getProperty(Prop.TWO_THETA).isDefault:
+            twoTheta = self.getProperty(Prop.TWO_THETA).value
```

Comment:
I don't think this 2theta stuff belongs here. If we passed `TWO_THETA` to the loader it would write the correct value to the logs readily. Also, why does a function called `_addSampleLogInfo` return 2theta?

Reply:
False?  First loading with user line position, but if the user does not provide a `BeamCentre` input, fitting takes place. Then, the two theta angle needs to be overwritten accordingly.